### PR TITLE
[Polyfill] add BATCH_TO_SPACE_ND and TRANSPOSE

### LINKED ISSUE #574: [API] Support TRANSPOSE op
Definition: https://android.googlesource.com/platform/frameworks/ml/+/master/nn/runtime/include/NeuralNetworks.h#2142

### LINKED ISSUE #576: [API] Support BATCH_TO_SPACE_ND op
Definition: https://android.googlesource.com/platform/frameworks/ml/+/master/nn/runtime/include/NeuralNetworks.h#1830

### PR DESCRIPTION
**Pull request by GreyZzzzzzXh**
fix #574, fix #576 

@huningxin @ibelem @Wenzhao-Xiang @pinzhenx , please help review, thanks.

### CODE REVIEW

**Comment by huningxin** [comment]
Thanks for the PR! I notice the https://github.com/intel/webml-polyfill/issues/575 is also needed by super resolution example (https://github.com/intel/webml-polyfill/pull/475). Is it good to include it in this PR?

File: src/nn/Enums.js [hunk 1]
```diff
@@ -1178,6 +1178,62 @@ export const OperationCode = {
...
+   * Supported tensor rank: 4, with "NHWC" or "NCHW" data layout.
+   * With the default data layout NHWC, the data is stored in the order of:
+   * [batch, height, width, channels]. Alternatively, the data layout could
+   * be NCHW, the data storage order of: [batch, channels, height, width].
+   *
+   * Inputs:
+   * * 0: An n-D tensor, specifying the tensor to be reshaped.
+   * * 1: A 1-D Tensor of {@link TENSOR_INT32}, the block sizes for each 
+   *      spatial dimension of the input tensor. All values must be >= 1.
+   * * 2: An optional {@link BOOL} scalar, default to false.
```
Comment (GreyZzzzzzXh) [inline src/nn/Enums.js]: This scalar is available since NNAPI level 29. Should it be defined as an INT32 if we remain this optional scalar?

**Comment by GreyZzzzzzXh** [comment]
@huningxin , TANH has been defined in [webml-polyfill/src/nn/Enums.js](https://github.com/intel/webml-polyfill/blob/master/src/nn/Enums.js#L1179). I will close #575 and open issues about its implementation on polyfill.

File: src/nn/Enums.js (hunk 1 above)
Comment (huningxin) [reply src/nn/Enums.js]: How about we introduce the `BOOL` type?

**Comment by huningxin** [comment]
> @huningxin , TANH has been defined in [webml-polyfill/src/nn/Enums.js](https://github.com/intel/webml-polyfill/blob/master/src/nn/Enums.js#L1179). I will close #575 and open issues about its implementation on polyfill.

Got it!

File: src/nn/Enums.js (hunk 1 above)
Comment (pinzhenx) [reply src/nn/Enums.js]: I agree with introducing `BOOL`. It's required by new `CONV` op, or we cannot distinguish implicit/exlicit padding by determine the number of inputs.

Comment (GreyZzzzzzXh) [reply src/nn/Enums.js]: Got it.

Comment (huningxin) [reply src/nn/Enums.js]: As discussion, we will handle the NCHW/NHWC in following PR.

**Comment by GreyZzzzzzXh** [comment]
@huningxin , PTAL, thanks.

**Comment by huningxin** [comment]
Looks good. Thanks!

**Pull request merged**